perfkitbenchmarker/network.py

```diff
@@ -22,20 +22,41 @@
 
 import threading
 
+from perfkitbenchmarker import context
+from perfkitbenchmarker import errors
+
 
 class BaseFirewall(object):
   """An object representing the Base Firewall."""
 
+
   # Lock used to serialize the instantiation of new BaseFirewall objects.
   _class_lock = threading.Lock()
 
   @classmethod
-  def GetFirewall(cls, firewalls_dict):
-    """Returns the Firewall."""
+  def GetFirewall(cls, *args, **kwargs):
+    """Returns a BaseFirewall.
+
+    This method is used instead of directly calling the class's constructor.
+    It creates BaseFirewall instances and registers them with the
+    BenchmarkSpec. If a BaseFirewall object has already been registered
+    with the same args and kwargs passed into this method, that object
+    will be returned rather than creating a new one. This enables multiple
+    VMs to call this method and all share the same BaseFirewall object.
+
+    Args:
+      args: The args to pass to the class's constructor.
+      kwargs: The kwargs to pass to the class's constructor.
+    """
+    benchmark_spec = context.GetThreadBenchmarkSpec()
+    if benchmark_spec is None:
+      raise errors.Error('GetFirewall called in a thread without a '
```

Comment:
Perhaps roll this error into a version of GetThreadBenchmarkSpec to be used when one is expected to exist, since that may be a common thing to complain about.